### PR TITLE
Added shield stats

### PR DESCRIPTION
**Pull request by ererbe**
# Description
More stats for the shields for more freedom in creating new shields (and the description)

- stats for resistance agianst: bullet,explosion and energy damage
- different "dont get hit time" until the shield starts to regenerate
- different shield regeneration speed

# Testing
Just copy the files to yours and play 

# Outstanding Work
- [x] Maybe better field for description (scrollable or something like this)

# Notes
> This is my first pull request on github... my englisch is may not correct

### CODE REVIEW

**Comment by GooeyHub** [comment]
Can one of the admins please verify this patch?

File: engine/src/main/java/org/destinationsol/game/item/Shield.java [hunk 1]
```diff
@@ -120,8 +124,14 @@ public void absorb(SolGame game, float dmg, Vector2 pos, SolShip ship, DmgType d
         }
         myIdleTime = 0f;
         if (dmgType == DmgType.BULLET) {
-            dmg *= BULLET_DMG_FACTOR;
+            dmg *= bulletDmgFactor;
         }
```
Comment (vampcat) [inline engine/src/main/java/org/destinationsol/game/item/Shield.java]: Can we convert these if conditions into an if else ladder?
```
if (dmgType == DmgType.BULLET) {
    ...
} else if (...) {
```

**Comment by vampcat** [comment]
@GooeyHub: ok to test

**Comment by GooeyHub** [comment]
Hooray Jenkins reported success with all tests good!

**Comment by GooeyHub** [comment]
Hooray Jenkins reported success with all tests good!

**Comment by ererbe** [comment]
If() is secound and third time now else if().
(Need I to commend every time I changed something?)

**Comment by vampcat** [comment]
Sorry for the delay.

No, you don't usually need to comment for the changes, relevant parties get mails everytime you make any changes to your PR (like push a commit). However, due to the GCI flood, this mail got lost somewhere.

I'm testing it right now.

**Comment by vampcat** [comment]
I'm not sure what's wrong, but with the PR, the shields don't work anymore. I loaded the game without your PR and chose Loaded Imperial Large (which has some good shields installed by default), and went to some guy and got hit - a purple-ish shield graphic came up, and my shield dropped from 40 to 38 to 36 and so on. With your PR, the graphic is missing and the value never goes below 40. You might want to see why that is. I'll also try and figure out what the problem is, and let you know if I figure something out.

**Comment by ererbe** [comment]
I made a mistake (but what exactly) this part makes the dmg go to 0...  i dont know why
`        if (dmgType == DmgType.BULLET) {
            dmg *= bulletDmgFactor;
        }
	else if (dmgType == DmgType.ENERGY) {
		dmg *= energyDmgFactor;
	}
	else if (dmgType == DmgType.EXPLOSION) {
		dmg *= explosionDmgFactor;
	}`

**Comment by vampcat** [comment]
@ererbe Just got time to finally debug this properly, since I was swamped with [Google Code-In](https://codein.withgoogle.com/) till now.

Either way, the reason was simple - you are loading the DmgFactors into myConfig.DmgFactors, but using DmgFactors from the class itself (which was never initialized). Changing it to `myConfig.DmgFactors` fixed the problem perfectly. I also removed the redundant DmgFactors from the class.

With that, this PR looks good to go. Thanks a lot for the changes :)
Merging, and adding you to contributors.

**Pull request merged**